subprojects/testing-jvm/src/main/java/org/gradle/api/internal/tasks/testing/detection/DefaultTestClassScanner.java

```diff
@@ -23,13 +23,14 @@
 import org.gradle.api.internal.tasks.testing.TestClassProcessor;
 import org.gradle.api.internal.tasks.testing.TestClassRunInfo;
 
-import java.io.File;
+import java.util.regex.Pattern;
 
 /**
  * The default test class scanner. Depending on the availability of a test framework detector,
  * a detection or filename scan is performed to find test classes.
  */
 public class DefaultTestClassScanner implements Runnable {
```

Comment:
Again, to keep the scanner simple, I would do this with a decorating test class processor.